pkg/api/handlers/compat/events.go

```diff
@@ -48,14 +48,27 @@ func GetEvents(w http.ResponseWriter, r *http.Request) {
 	}()
 	if eventsError != nil {
 		utils.InternalServerError(w, eventsError)
+		close(eventChannel)
 		return
 	}
 
-	coder := json.NewEncoder(w)
-	coder.SetEscapeHTML(true)
+	// If client disappears we need to stop listening for events
+	go func(done <-chan struct{}) {
+		<-done
+		close(eventChannel)
+	}(r.Context().Done())
 
+	// Headers need to be written out before turning Writer() over to json encoder
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
+	if flusher, ok := w.(http.Flusher); ok {
+		flusher.Flush()
+	}
+
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
```

Comment:
Why are we doing this inside one function? This should be done once, for the entire package, at global scope

Reply:
@mheon I'm using the `json` variable as an shortcut to the variable `jsoniter.ConfigCompatibleWithStandardLibrary`. What is the benefit of a global scoped variable? 

I just substituted jsoniter for json package in this case.  If there is some benefit of a global variable I'll be happy to cover that in a follow up PR.

Reply:
Global scope replaces all uses of `encoding/json` with `jsoniter` within the packages - which, given its performance benefits and compatibility with the standard library, is desirable. We're doing this within Libpod with great success.